r/tests/testthat/test-dplyr-arrange.R

```diff
@@ -151,13 +153,50 @@ test_that("arrange", {
     "not supported in Arrow",
     fixed = TRUE
   )
+})
+
+test_that("arrange() on datetime columns", {
+  expect_dplyr_equal(
+    input %>%
+      arrange(dttm, int) %>%
+      collect(),
+    tbl
+  )
+  skip("Sorting by only a single timestamp column fails (ARROW-12087)")
+  expect_dplyr_equal(
+    input %>%
+      arrange(dttm) %>%
+      collect(),
+    tbl %>%
+      select(dttm, grp)
+  )
+})
+
+test_that("arrange() on logical columns", {
+  skip("Sorting by bool columns is not supported (ARROW-12016)")
+  expect_dplyr_equal(
+    input %>%
+      arrange(lgl, int) %>%
+      collect(),
+    tbl
+  )
+})
+
+test_that("arrange() with bad inputs", {
   expect_error(
     tbl %>%
       Table$create() %>%
       arrange(1),
     "does not contain any field names",
     fixed = TRUE
   )
+  expect_error(
+    tbl %>%
+      Table$create() %>%
+      arrange(aertidjfgjksertyj),
+    "not found",
```

Comment:
For safety, this should probably be wrapped in `with_language("en", ...)` though testthat/check do seem to set LANGUAGE="en" now.

Reply:
Done in dd2e0efd05ff46fae184ebb18965a39cdbfcce96

Reply:
Sorry, on second thought this is not necessary because we set the language in the setup (probably right by where you added the LC_COLLATE 🤦 ) https://github.com/apache/arrow/blob/master/r/tests/testthat/helper-arrow.R#L35

But it's fine to leave it in, it won't hurt anything.

Reply:
I'll remove it...